server/memphis_handlers_user_mgmt.go

```diff
@@ -651,14 +677,27 @@ func (umh UserMgmtHandler) AddUser(c *gin.Context) {
 
 	var brokerConnectionCreds string
 	if userType == "application" {
-		brokerConnectionCreds, err = AddUser(username)
-		if err != nil || len(username) == 0 {
-			serv.Errorf("CreateUser: User " + body.Username + ": " + err.Error())
-			c.AbortWithStatusJSON(500, gin.H{"message": err.Error()})
-			return
+		if configuration.USER_PASS_BASED_AUTH {
+			if body.Password == "" {
+				serv.Warnf("CreateUser: Password was not provided for user " + username)
+				c.AbortWithStatusJSON(SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": "Password was not provided"})
+				return
+			}
+			password = body.Password
+			avatarId = 1
+			if body.AvatarId > 0 {
+				avatarId = body.AvatarId
+			}
+		} else {
+			brokerConnectionCreds, err = AddUser(username)
```

Comment:
turn the add user function to be smarter. I mean it should decide on how to create the user and to make the reload